### ISSUE TITLE
[question] How to build a Library which has it self with another Architecture as Requirement?

### ISSUE THREAD

**Issue by werto87** [conan-io/conan#9440]
I want to build corrade.  Host is emscripten and build is linux.
 The documentation says:

> Native build of corrade-rc
For all cross-compilation except the UWP / Windows RT build, you need to have a native version of Corrade installed, because Corrade needs to run the corrade-rc utility on the host system as part of the build process. If native version of corrade-rc is not found on the system, cross-compilation will fail. If corrade-rc is not in PATH, point CMake to it using -DCORRADE_RC_EXECUTABLE=/path/to/corrade-rc.

[I updated the corrade conan recipe](https://github.com/werto87/conan_corrade/blob/master/all/conanfile.py) and I can build corrade for emscripten but I need to install corrade on my system. I want to use conan to build a corrade package with linux as host and then set the path to corrade-rc and build a corrade package with emscripten as host.

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @werto87 

At the moment cyclic dependencies, including bootstrapping are not allowed in dependency graphs. In general, they don't make sense, and the particular case of bootstrapping was not enough requested so far to be considered.

This will probably be changed in Conan 2.0, and there is exploratory code already there, to be able to have such bootstrapping for ``build`` dependencies only.

**Comment by memsharded** [comment]
This test in https://github.com/conan-io/conan/blob/4f2f8bbddebf04b500076edcb5a207a5d3d8f1ed/conans/test/integration/build_requires/test_install_test_build_require.py#L182 proofs that it is possible to depend on yourself with:

```python
 class Pkg(ConanFile):
     name = "tool"
     version = "1.0"
     settings = "os"
     def build_requirements(self):
          if cross_building(self):
             self.tool_requires("tool/1.0")
```

Closing this as solved in 2.0
